File: packages/main-api/src/features/spaces/rewards/types/reward_action.rs
use std::{fmt::Display, str::FromStr};

use serde_with::{DeserializeFromStr, SerializeDisplay};

use crate::*;

// RewardAction is available actions of a reward
// Like Poll, Quiz, etc

#[derive(
    Debug,
    Clone,
    DynamoEnum,
    SerializeDisplay,
    DeserializeFromStr,
    JsonSchema,
    OperationIo,
    Eq,
    PartialEq,
    Default,
)]
pub enum RewardAction {
    #[default]
    Poll,
    // Quiz,
}

impl TryFrom<&EntityType> for RewardAction {
    type Error = Error;
    fn try_from(entity_type: &EntityType) -> Result<Self> {
        match entity_type {
            EntityType::SpacePoll(_) => Ok(RewardAction::Poll),
            _ => Err(Error::InvalidEntityType),
        }
    }
}

File: ts-packages/web/src/features/spaces/rewards/types/reward-action.ts
export enum RewardAction {
  Poll = 'Poll',
  // Quiz = 'Quiz',
}

import { RewardUserBehavior } from './reward-user-behavior';

export function getRewardUserBehaviorFromAction(
  action: RewardAction,
): RewardUserBehavior[] {
  switch (action) {
    case RewardAction.Poll:
      return [RewardUserBehavior.RespondPoll];
    default:
      return [];
  }
}
